traitsui/testing/tester/ui_tester.py

```diff
@@ -217,83 +217,18 @@ def find_by_id(self, ui, id):
         ).find_by_id(id=id)
 
 
-def _get_editor_by_name(ui, name):
-    """ Return a single Editor from an instance of traitsui.ui.UI with
-    a given extended name. Raise if zero or many editors are found.
-
-    Parameters
-    ----------
-    ui : traitsui.ui.UI
-        The UI from which an editor will be retrieved.
-    name : str
-        A single name for retrieving an editor on a UI.
-
-    Returns
-    -------
-    editor : Editor
-        The single editor found.
-    """
-    editors = ui.get_editors(name)
-
-    all_names = [editor.name for editor in ui._editors]
-    if not editors:
-        raise ValueError(
-            "No editors can be found with name {!r}. "
-            "Found these: {!r}".format(name, all_names)
-        )
-    if len(editors) > 1:
-        raise ValueError(
-            "Found multiple editors with name {!r}.".format(name))
-    editor, = editors
-    return editor
-
-
-def _get_editor_by_id(ui, id):
-    """ Return single Editor from an instance of traitsui.ui.UI with
-    the given identifier.
-
-    Parameters
-    ----------
-    ui : traitsui.ui.UI
-        The UI from which an editor will be retrieved.
-    id : str
-        Id for finding an item in the UI.
-
-    Returns
-    -------
-    editor : Editor
-        The single editor found.
-    """
-    try:
-        editor = getattr(ui.info, id)
-    except AttributeError:
-        raise ValueError(
-            "No editors found with id {!r}. Got these: {!r}".format(
-                id, ui._names)
-            )
-    return editor
-
-
 def _get_ui_registry():
     """ Return a TargetRegistry with traitsui.ui.UI as the target.
 
     Parameters
     ----------
     registry : TargetRegistry
     """
+
     registry = TargetRegistry()
-    registry.register_solver(
-        target_class=UI,
-        locator_class=locator.TargetByName,
-        solver=lambda wrapper, location: (
-            _get_editor_by_name(wrapper.target, location.name)
-        ),
-    )
-    registry.register_solver(
+    register_nested_ui_solvers(
+        registry=registry,
         target_class=UI,
-        locator_class=locator.TargetById,
-        solver=lambda wrapper, location: (
-            _get_editor_by_id(wrapper.target, location.id)
-        ),
+        nested_ui_getter=lambda target: target,
     )
     return registry
```

Comment:
It is not a workaround. It might look like a workaround because the function is named `register_nested_ui_solvers` (with the word "nested" in it)? I can rename the function in a separate PR.

Reply:
Maybe `register_traitsui_ui_solvers`, considering the word "ui" is so overloaded.

Reply:
And then `nested_ui_getter` to `traitsui_ui_getter` too.